Firebase/Database/Api/FIRDatabase.m

```diff
@@ -69,6 +77,23 @@ + (void)load {
           }
       }
   }];
+  [FIRComponentContainer registerAsComponentRegistrant:self];
+}
+
++ (NSArray<FIRComponent *> *)componentsToRegister {
+    FIRDependency *auth =
+        [FIRDependency dependencyWithProtocol:@protocol(FIRAuthInterop)
+                                   isRequired:NO];
+    FIRComponentCreationBlock creationBlock =
+        ^id _Nullable(FIRComponentContainer *container, BOOL *isCacheable) {
+        return [self databaseForApp:container.app];
```

Comment:
Let's chat about this when I get in, but I think we should follow the Storage PR https://github.com/firebase/firebase-ios-sdk/pull/1660 which will also move the instance management to the container. It'd also help keep this class a bit cleaner without much knowledge of the component system and just inject the `auth` instance as needed.

Reply:
I can make a component class like `FIRStorageComponent` and remove as many references to the `FIRApp` as possible from the initializers in `FIRDatabase`, only passing around the `auth` instance instead.

Is this what you mean?

Reply:
This has all moved to the new `FIRDatabaseProvider` class.